### PR TITLE
Update to Kubernetes 1.16.2

### PR DESCRIPTION
**Pull request by surajssd**
- Dependent on https://github.com/kinvolk/terraform-render-bootkube/pull/23
- Update etcd to v3.4.2
- Update the provider versions
- Update the terraform ct provider to 0.4.0
- Update Kubernetes to 1.16.2
- Update Calico to 3.9.2
- Remove labels `node-role.kubernetes.io`, kubelet cannot set these labels using the flag --node-labels. This is enforced as a security feature. But this can be set manually by cluster-admin.

### CODE REVIEW

File: aws/flatcar-linux/kubernetes/cl/controller.yaml.tmpl
Comment (invidian) [inline aws/flatcar-linux/kubernetes/cl/controller.yaml.tmpl]: You sure etcd 3.4.1 is supported? https://github.com/kubernetes/kubernetes/issues/82641

Comment (surajssd) [reply aws/flatcar-linux/kubernetes/cl/controller.yaml.tmpl]: I think k8s community is in general is slow with adoption of newer base components because they need to be changed in various different locations. And it is a task in itself.

**Comment by surajssd** [comment]
Dependent on https://github.com/kinvolk/terraform-render-bootkube/pull/23

File: aws/flatcar-linux/kubernetes/cl/controller.yaml.tmpl
Comment (dongsupark) [reply aws/flatcar-linux/kubernetes/cl/controller.yaml.tmpl]: FYI, I have been testing etcd v3.4.1 while I was testing my ARM support branch https://github.com/kinvolk/lokomotive-kubernetes/pull/78. (Funny enough, I had to upgrade etcd to v3.4.1 because of container images for etcd v3.3.x did not work with rkt on Flatcar)
So far it was ok for me, no problem at all with etcd v3.4.
Though I'm testing only on Packet.
Actually I'm not sure if any corner case can pop up when upgrading etcd to v3.4.

Comment (rata) [reply aws/flatcar-linux/kubernetes/cl/controller.yaml.tmpl]: Typhoon seems to be using 3.4.2 (released one week ago)

File: azure/flatcar-linux/kubernetes/bootkube.tf
Comment (rata) [inline azure/flatcar-linux/kubernetes/bootkube.tf]: nit pick, but shouldn't this be the merge commit now that is merged? I mean 7e9cf2460a3f01703f2edbfde463c1843fbf8d49

**Review by rata — DISMISSED** [review]
LGTM, didn't test it myself. I trust the CI :see_no_evil: 

Just a minor comment on the bootkube commit used.

Oh, one **important** thing! 1.16 removes support for several deprecated stuff (https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/). Are we really sure that all compents that are around have been updated? Maybe makes sense to make sure they are tested too...

We probably should **NOT** merge until we are really sure that all the removals in 1.16 don't break anything

**Review by rata — COMMENTED** [review]
Click approved and later realized about the 1.16. I tried to removed the approve but couldn't :-D. EDIT: found it :-D

I want to emphazize we should not merge this until we are absolutely sure that it doesn't break other things using the deprecated stuff. We do not want to have a master we can't use. I'd like to really check that everything works with this before approval :-D

File: azure/flatcar-linux/kubernetes/bootkube.tf
Comment (surajssd) [reply azure/flatcar-linux/kubernetes/bootkube.tf]: done!

**Comment by surajssd** [comment]
@rata 

> I want to emphazize we should not merge this until we are absolutely sure that it doesn't break other things using the deprecated stuff. We do not want to have a master we can't use. I'd like to really check that everything works with this before approval :-D


What do you mean by other things? I have tested all the components and made necessary upgrades to them.

**Comment by rata** [comment]
> @rata
> What do you mean by other things? I have tested all the components and made necessary upgrades to them.

Perfect! This is what I meant! Just make sure all runs fine before merging (don't want to have a broken master). That is all :)

**Pull request merged**
